File: api-server/internal/model/notification.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
)

// Notification 系统全局通知消息
type Notification struct {
	// 通知唯一ID
	ID uuid.UUID `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	// 通知标题（最大100字符）
	Title string `gorm:"not null;size:100" json:"title"`
	// 通知正文
	Content string `gorm:"not null;type:text" json:"content"`
	// 是否已读
	IsRead bool `gorm:"not null;default:false;index" json:"is_read"`
	// 告警级别：critical/high/medium/low/info
	Severity string `gorm:"not null;default:'info';size:20" json:"severity"`
	// 通知类型：rule_generated/alert_triggered/approval_required/system
	Type string `gorm:"not null;size:50;index" json:"type"`
	// 可选跳转路径
	Link string `gorm:"size:500" json:"link,omitempty"`
	// 扩展业务字段（JSON）：rule_id/mitre_id/trigger_count/alert_ids 等
	Metadata datatypes.JSON `gorm:"type:jsonb" json:"metadata,omitempty"`
	// 创建时间
	CreatedAt time.Time `gorm:"not null;index:,sort:desc" json:"timestamp"`
	// 更新时间
	UpdatedAt time.Time `gorm:"not null" json:"-"`
}

// TableName 返回表名
func (Notification) TableName() string {
	return "notifications"
}

File: frontend/src/types/notification.ts
// 通知严重级别
export type NotificationSeverity = 'critical' | 'high' | 'medium' | 'low' | 'info'

// 通知类型
export type NotificationType =
  | 'rule_generated'       // AI 规则自动生成
  | 'alert_triggered'      // 告警触发
  | 'approval_required'    // 待审核规则
  | 'system'               // 系统消息

// 通知元数据（业务扩展字段）
export interface NotificationMetadata {
  i18n_key?:     string
  i18n_params?:  Record<string, string | number | boolean | null>
  rule_id?:       string    // 关联规则 ID（rule_generated 时携带）
  mitre_id?:      string    // 关联 MITRE ID
  trigger_count?: number    // 触发次数
  trigger_hours?: number    // 触发时间窗口（小时）
  alert_ids?:     string[]  // 关联告警 ID 列表
  host_ids?:      string[]  // 关联主机 ID 列表
}

// 通知主体
export interface Notification {
  id:        string
  title:     string
  content:   string
  is_read:   boolean
  timestamp: string                   // ISO 8601
  severity:  NotificationSeverity
  type:      NotificationType
  link?:     string
  metadata?: NotificationMetadata
}

// 列表响应
export interface NotificationListResponse {
  list:         Notification[]
  total:        number
  unread_count: number
  page:         number
  page_size:    number
}

// 标记已读响应
export interface MarkReadResponse {
  success: boolean
}

// 全部标记已读响应
export interface MarkAllReadResponse {
  success: boolean
  updated_count: number
}
